### PR TITLE
RS and RC: Client-side caching and compatibility

### PR DESCRIPTION
**Pull request by rrelledge**
DOC-3951

Please note that some of the relrefs that point to client-side caching devex docs will not work until Andy's PR https://github.com/redis/docs/pull/530 is merged or cherry-picked to this branch. I thought it might be easier to review the Enterprise-specific changes if I waited to pull in Andy's changes.

### CODE REVIEW

**Review by yoavredis — APPROVED** [review]
@rrelledge LGTM, 10x!

File: content/operate/rc/compatibility.md [hunk 1]
```diff
@@ -19,6 +19,10 @@ Both [Redis Enterprise Software]({{< relref "/operate/rs" >}}) and Redis Cloud a
 
 Redis Enterprise Software and Redis Cloud support RESP2 and RESP3. In Redis Cloud, you can choose between RESP2 and RESP3 when you [create a database]({{< relref "/operate/rc/databases/create-database" >}}) and you can change it when you [edit a database]({{< relref "/operate/rc/databases/view-edit-database" >}}). For more information about the different RESP versions, see the [Redis serialization protocol specification]({{< relref "/develop/reference/protocol-spec" >}}#resp-versions).
 
+## Client-side caching compatibility
+
+Redis Software and Redis Cloud support client-side caching for databases with Redis versions 7.4 or later. See [Client-side caching compatibility with Redis Software and Redis Cloud]({{<relref "/operate/rs/references/compatibility/client-side-caching">}}) for more information about compatibility.
```
Comment (andy-stark-redis) [inline content/operate/rc/compatibility.md:24]:
```suggestion
Redis Software and Redis Cloud support [client-side caching]({{<relref "/develop/connect/clients/client-side-caching">}}) for databases with Redis versions 7.4 or later. See [Client-side caching compatibility with Redis Software and Redis Cloud]({{<relref "/operate/rs/references/compatibility/client-side-caching">}}) for more information about compatibility.
```

File: content/operate/rs/references/compatibility/_index.md [hunk 2]
```diff
@@ -20,6 +20,10 @@ Both Redis Enterprise Software and [Redis Cloud]({{< relref "/operate/rc" >}}) a
 
 Redis Enterprise Software and Redis Cloud support RESP2 and RESP3. See [RESP compatibility with Redis Enterprise]({{< relref "/operate/rs/references/compatibility/resp" >}}) for more information.
 
+## Client-side caching compatibility
+
+Redis Software and Redis Cloud support client-side caching for databases with Redis versions 7.4 or later. See [Client-side caching compatibility with Redis Software and Redis Cloud]({{<relref "/operate/rs/references/compatibility/client-side-caching">}}) for more information about compatibility and configuration options.
```
Comment (andy-stark-redis) [inline content/operate/rs/references/compatibility/_index.md:25]:
```suggestion
Redis Software and Redis Cloud support [client-side caching]({{<relref "/develop/connect/clients/client-side-caching">}}) for databases with Redis versions 7.4 or later. See [Client-side caching compatibility with Redis Software and Redis Cloud]({{<relref "/operate/rs/references/compatibility/client-side-caching">}}) for more information about compatibility and configuration options.
```

File: content/operate/rs/references/compatibility/client-side-caching.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+| Redis product           | Client-side caching support |
+|-------------------------|-----------------------------|
+| Redis Community Edition | Redis v6.0 and later |
+| Redis Cloud             | Redis database v7.4 and later |
+| Redis Software          | Redis database v7.4 and later |
+
+## Supported RESP versions
+
+Client-side caching in Redis Software and Redis Cloud requires RESP3.
```
Comment (andy-stark-redis) [inline content/operate/rs/references/compatibility/client-side-caching.md:30]: Not sure if this is really necessary but thought I'd suggest it. (The linked page might be too technical anyway.)
```suggestion
Client-side caching in Redis Software and Redis Cloud requires [RESP3]({{< relref "/develop/reference/protocol-spec#resp-versions" >}}).
```

**Review by andy-stark-redis — APPROVED** [review]
Just a few suggestions for some links, but otherwise excellent :-)

**Comment by mortensi** [comment]
Looks good to me. What happens if tracking-table-max-keys is exceeded? @yoavredis

**Pull request merged**